_projects/coala_language_server.md

```diff
@@ -0,0 +1,63 @@
+---
+collaborating_projects:
+  - coala
+desc: "Offer coala languager server."
+difficulty: medium
+initiatives:
+  - GSoC
+  - RGSoC
+issues:
+  - "https://github.com/coala/coala-vs-code/issues/3"
+  - "https://github.com/coala/coala-vs-code/issues/4"
+markdown: coala_language_server.md
+mentors:
+  - gaocegege
+name: "coala Language Server"
+requirements:
+  - "The applicant should have at least one patch accepted to any of the coala."
+  - "The applicant shoule be familiar with language server protocol"
+tags:
+  - Language Server
+  - Plugins
+---
+
+The [Language Server Protocol](https://langserver.org/) was created by Microsoft to define a common language for programming language analyzers to speak. Today, several companies have come together to support its growth, including Codenvy, Red Hat, and Sourcegraph, and the protocol is becoming supported by a rapidly growing list of editor and language communities:
+
+- VS Code
+- Eclipse Che
+- Eclipse IDE(Eclipse LSP4E)
+- IntelliJ / JetBrains IDEs
+- Vim
+- Atom (WIP)
+- Emacs (WIP)
+- Sublime (WIP)
+
+We have a state-of-art implementation in [coala-vs-code](https://github.com/coala/coala-vs-code) but it is designed for VS Code only. We need to design and implement a general mechanism for multiple editors/IDEs support.
+
+The project is divided into two parts:
+
+1. Offer a basic languager server.
+2. Implement a plugin based on the language server.
+
+#### Milestones
+
+##### PREPARATION/BONDING
+
+- An official repository exists with CI.
+- An mock language server exists.
```

Comment:
what is in this 'mock' server?

Reply:
It means that we could see the interactions between the server and frontend, but it can do nothing but print.

Reply:
I'm not sure everybody will understand this. I think you can write it in the description to be clear for students

Reply:
Ah, this is a separate deliverable. I think this should be called a stub LS.

This is a very nice idea, and it should be a repository that is left with only a stub LS, so it is easy for other people to use as a basis for different projects.

This should be quite easy , as it could be extracted from coala-vs-code , and other python LS implementations.